### DISCUSSION TITLE
qEditor: paste from Word

### DISCUSSION THREAD

**Discussion by robgwin** [quasarframework/quasar#16158] [General - Components / Directives / etc] [answered]
Wondering if anyone's got a good way to deal with people pasting MS Word content directly into a qEditor component? Something like CKEditor's Paste From Word feature perhaps? I've been trying to strip the unwanted tags with DomParser but it's very difficult to accommodate all possible scenarios and it feels like reinventing the wheel. Any thoughts most welcome!

**Comment by shengrass** [comment]
I have a same issue, does anyone can help?

**Comment by robgwin** [comment, marked as answer]
In my case I discovered that (a) most of my problems were caused by the huge code blocks MS Word injects in html comments, and (b) I can intercept a paste with the paste.native event and insertText command. So this is what I ended up with:

```vue
<template>
  <q-editor
    v-model="myContent"
    ref="myEditor"
    @paste.native="evt => pasteCleanHtml(evt, 'myEditor')"
  />
</template>
<script>
export default {
  methods: {
    pasteCleanHtml(evt, editor_refname) {
      // get the clipboard
      const pasted = evt.clipboardData.getData('text')

      // leave plain text alone
      if (!pasted || pasted[0] !== '<') return

      // stop the paste
      evt.preventDefault()

      // strip out html comments (mainly to eliminate lengthy MS Word junk like <!--[if gte mso 9]> etc)
      const cleaned = pasted.replace(/<\!--.*?-->/gs, "");

      // could do more things here?

      // paste it
      this.$refs[editor_refname].runCmd('insertText', cleaned)
    }
  }
}
```

**Reply by shengrass** [reply]
Hello robgwin, thank you for your reply and give me the sample code.

Have you ever tried that copy MS Word conetnt with image?

When I copy text and image from MS Word and paste to QEditor, the image is not displayed.

Would you have any idea? Thank you so much.

![螢幕擷取畫面 2023-11-27 172109](https://github.com/quasarframework/quasar/assets/27479292/419b0bb9-4d53-4c84-ad99-54933bc14314)

![螢幕擷取畫面 2023-11-27 172637](https://github.com/quasarframework/quasar/assets/27479292/a7b87586-2af5-4eed-929e-6eea081bc3c1)

**Reply by robgwin** [reply]
That's not really a Word-specific problem. Image handling is a whole other challenge of its own. You need to figure out a way to read the local image file and either host it somewhere or embed as data. Good luck.
